.github/workflows/build-push.yml

```diff
@@ -0,0 +1,289 @@
+      GITHUB_USER_NAME:
+        description: Username for the GitHub user configuration.
+        required: true
+      GITHUB_USER_SSH_KEY:
+        description: Private SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: true
+      GITHUB_USER_SSH_PUBLIC_KEY:
+        description: Public SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: true
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  build-push:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    env:
+      NO_CHANGES: ''                                   # we'll override if no changes to commit
+      # Disables symlinking of local "path" repositories during package installation.
+      # During development, symlinking is preferable.
+      # In production builds, we ship the actual install location and remove the source directory.
+      COMPOSER_MIRROR_PATH_REPOS: 1
+      GIT_SHA: ${{ github.sha }}
+      PACKAGE_VERSION: ${{ inputs.PACKAGE_VERSION }}  # Will be overridden by determine package version step
+      PACKAGE_FOLDER_NAME: ${{ inputs.PACKAGE_FOLDER_NAME && inputs.PACKAGE_FOLDER_NAME || github.event.repository.name }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ssh-key: ${{ secrets.GITHUB_USER_SSH_KEY }}
+
+      - name: Set up SSH
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.GITHUB_USER_SSH_KEY }}
+
+      - name: Set up Git
+        run: |
+          git config --global user.email "${{ secrets.GITHUB_USER_EMAIL }}"
+          git config --global user.name "${{ secrets.GITHUB_USER_NAME }}"
+          git config --global advice.addIgnoredFile false
+          git config --global push.autoSetupRemote true
+
+      - name: Set up signing commits
+        run: |
+          : # Create empty SSH private key file so Git does not complain.
+          touch "${{ runner.temp }}/signingkey"
+          echo "${{ secrets.GITHUB_USER_SSH_PUBLIC_KEY }}" > "${{ runner.temp }}/signingkey.pub"
+          git config --global commit.gpgsign true
+          git config --global gpg.format ssh
+          git config --global user.signingkey "${{ runner.temp }}/signingkey.pub"
+
+      - name: Set up custom environment variables
+        env:
+          ENV_VARS: ${{ secrets.ENV_VARS }}
+        if: ${{ env.ENV_VARS }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+
+      - name: Determine package version
+        run: |
+          if [ -n "${{ inputs.PACKAGE_VERSION }}" ]; then
+            PACKAGE_VERSION="${{ inputs.PACKAGE_VERSION }}"
+            echo "Using provided package version: $PACKAGE_VERSION"
```

Comment:
Is this "if" required? The `env.PACKAGE_VERSION` is already set above the steps. So maybe we can run this step with an `if:`-condition?